checkmate/url/hash.py

```diff
@@ -0,0 +1,37 @@
+"""Implementation of Google Web Risk URL hashing.
+
+As defined here: https://cloud.google.com/web-risk/docs/urls-hashing#hash_computations
+"""
+
+from hashlib import sha256
+
+from checkmate.url.canonicalize import CanonicalURL
+from checkmate.url.expand import ExpandURL
+
+# This is all very speculation based at this point, we're not sure how this
+# is going to look until we have our hands on some data.
+
```

Comment:
This code is super speculative. I'd basically ignore it for now. We aren't going to use it immediately, and I've no idea what shape it will take in the end.